### PR TITLE
feat: Session owner setting panel in Neo session launcher

### PR DESCRIPTION
**Pull request by yomybaby**
### TL;DR

Added session owner setter functionality and improved resource group selection.

image
image

### What changed?

- Introduced `SessionOwnerSetterCard` and `SessionOwnerSetterPreviewCard` components for setting session owners
- Added `BAISelect` component for enhanced select functionality
- Updated `ResourceGroupSelect` to use project name instead of ID
- Modified `AvailableResourcesCard` to use `ResourceGroupSelectForCurrentProject`
- Updated `SessionLauncherPage` to include session owner setter for admin and superadmin roles
- Added new words to the spell checker configuration
- Add ErrorBoundary with empty fallback to ignore `SessionLauncherValidationTour`'s error

### How to test?

1. Log in as an admin or superadmin user
2. Navigate to the session launcher page
3. Verify that the session owner setter card is visible
4. Test the functionality of setting a session owner:
   - Enter an owner's email
   - Select an access key
   - Choose a group and resource group
5. Confirm that the preview card updates correctly
6. Launch a session and verify that the owner settings are applied

### Why make this change?

This change enhances the session management capabilities for administrators, allowing them to launch sessions on behalf of other users.

### CODE REVIEW

**Comment by yomybaby** [comment]
* **#2693** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2693?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @yomybaby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: react/src/components/ResourceGroupSelect.tsx [hunk 1]
```diff
@@ -142,13 +139,13 @@ const ResourceGroupSelect: React.FC<ResourceGroupSelectProps> = ({
...
-        if (open) {
-          startLoadingTransition(() => {
-            updateFetchKey();
-          });
-        }
+        // if (open) {
+        //   startLoadingTransition(() => {
+        //     updateFetchKey();
+        //   });
+        // }
```
Comment (ironAiken2) [inline react/src/components/ResourceGroupSelect.tsx:146]: Is there a reason you commented this out?

**Review by ironAiken2 — DISMISSED** [review]
The User Email of the delegated session is showing as the email of the user who actually requested the creation.

Relatedly, there are cases where the user who has taken over the privileges is unable to use the control functions within the session list (running the terminal, checking logs, etc...), partly because email is registered as the creation requestor and not the user.

File: react/src/components/ResourceGroupSelect.tsx (hunk 1 above)
Comment (yomybaby) [reply react/src/components/ResourceGroupSelect.tsx:146]: The `ResourceGroup` does not change frequently. It is not necessary to refetch it every time the dropdown is opened.

**Review by agatha197 — APPROVED** [review]
LGTM

**Pull request merged**